packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart

```diff
@@ -558,6 +558,9 @@ class WebKitWebViewController extends PlatformWebViewController {
   Future<void> setInspectable(bool inspectable) {
     return _webView.setInspectable(inspectable);
   }
+
+  @override
+  Future<String?> getUserAgent() => _webView.getCustomUserAgent();
```

Comment:
These aren't equivalent; the RHS returns null unless a custom user agent has been set, so this implementation would give people the wrong answer on iOS (except in cases where they set it themselves, but they probably don't need to ask then). It looks like we need to fall back to using JS when this returns null.

Reply:
This issue also suggests we need an integration test that validates that we get a non-null value when we haven't set a custom agent.